base/ocsp/src/main/java/com/netscape/cms/ocsp/DefStore.java

```diff
@@ -88,12 +90,12 @@ public class DefStore implements IDefStore, IExtendedPluginInfo {
     private static final String PROP_INCLUDE_NEXT_UPDATE =
             "includeNextUpdate";
 
-    protected Hashtable<String, Long> mReqCounts = new Hashtable<>();
+    protected Map<String, Long> mReqCounts = new HashMap<>();
```

Comment:
If this was a local variable it would be safe to convert a `Hashtable` into a `Map` since that variable was only used by a single thread. However, since it's a field in a class, and the class seems to be used for OCSP validation, so theoretically there could be multiple threads accessing the field concurrently, so I'm not quite sure if it's safe. What do you think?

Reply:
This particular `Hashtable` seems to be used to count the requests received by each `CRLIssuingPointRecord`. Since the number of `CRLIssuingPointRecord` itself probably doesn't change a lot, it might be safe to convert it into a `Map`.

Reply:
Hmm, good point. If we're doing validation/reading then perhaps it is sufficient to make the map `volatile` rather than going for a full blown `synchronization`? That way we ensure any writes to the map are done before any attempts to read it, but don't go as far as blocking threads from accessing the map. What do you think?

Reply:
There are several things to consider here:
* the `mReqCounts` field which contains a reference to a `HashMap` object
* the keys in the `HashMap`
* the values in the `HashMap`

I'm not too familiar with `volatile` but based on what I read so far will it only apply to the `mReqCounts` field, but not to the keys/values in the map, so if the map is updated at the same time by multiple threads there is still a risk that the keys/values in the map (but not the field) might get corrupted.

As for the `mReqCounts` field itself, since the code does not update the value after the initialization there is actually no risk of field corruption.

As for the keys in the `HashMap`, as I said earlier most people probably will only have one key (i.e. the `MasterCRL`), so once that key is added in `incReqCount()` there will be no more changes to the hash table inside the `HashMap`, but it's possible that several OCSP clients might call the initial `incReqCount()` at the same time, so there's still a small risk for map key corruption.

As for the values in the `HashMap`, the process to increment the counter is not atomic, so there's also a (pretty big) risk of map value corruption, but I think this is actually an existing problem in the current code.

So for this PR I'd suggest to revert it back to `Hashtable` and merge it, then we can address the concurrency issue in a separate PR. We probably need to use a `HashMap` of `AtomicLong`.